models/webhook.go

```diff
@@ -481,20 +567,60 @@ type HookEventType string
-	HookEventPush                HookEventType = "push"
-	HookEventIssues              HookEventType = "issues"
-	HookEventIssueComment        HookEventType = "issue_comment"
-	HookEventPullRequest         HookEventType = "pull_request"
-	HookEventRepository          HookEventType = "repository"
-	HookEventRelease             HookEventType = "release"
-	HookEventPullRequestApproved HookEventType = "pull_request_approved"
-	HookEventPullRequestRejected HookEventType = "pull_request_rejected"
-	HookEventPullRequestComment  HookEventType = "pull_request_comment"
+	HookEventCreate                    HookEventType = "create"
+	HookEventDelete                    HookEventType = "delete"
+	HookEventFork                      HookEventType = "fork"
+	HookEventPush                      HookEventType = "push"
+	HookEventIssues                    HookEventType = "issues"
+	HookEventIssueAssign               HookEventType = "issue_assign"
+	HookEventIssueLabel                HookEventType = "issue_label"
+	HookEventIssueMilestone            HookEventType = "issue_milestone"
+	HookEventIssueComment              HookEventType = "issue_comment"
+	HookEventPullRequest               HookEventType = "pull_request"
+	HookEventPullRequestAssign         HookEventType = "pull_request_assign"
+	HookEventPullRequestLabel          HookEventType = "pull_request_label"
+	HookEventPullRequestMilestone      HookEventType = "pull_request_milestone"
+	HookEventPullRequestComment        HookEventType = "pull_request_comment"
+	HookEventPullRequestReviewApproved HookEventType = "pull_request_review_approved"
+	HookEventPullRequestReviewRejected HookEventType = "pull_request_review_rejected"
+	HookEventPullRequestReviewComment  HookEventType = "pull_request_review_comment"
+	HookEventPullRequestSync           HookEventType = "pull_request_sync"
+	HookEventRepository                HookEventType = "repository"
+	HookEventRelease                   HookEventType = "release"
 )
 
+// Event returns the HookEventType as an event string
+func (h HookEventType) Event() string {
+	switch h {
+	case HookEventCreate:
+		return "create"
+	case HookEventDelete:
+		return "delete"
+	case HookEventFork:
+		return "fork"
+	case HookEventPush:
+		return "push"
+	case HookEventIssues, HookEventIssueAssign, HookEventIssueLabel, HookEventIssueMilestone:
+		return "issues"
+	case HookEventPullRequest, HookEventPullRequestAssign, HookEventPullRequestLabel, HookEventPullRequestMilestone,
+		HookEventPullRequestSync:
+		return "pull_request"
+	case HookEventIssueComment, HookEventPullRequestComment:
+		return "issue_comment"
+	case HookEventPullRequestReviewApproved:
+		return "pull_request_approved"
+	case HookEventPullRequestReviewRejected:
+		return "pull_request_rejected"
+	case HookEventPullRequestReviewComment:
+		return "pull_request_comment"
+	case HookEventRepository:
+		return "repository"
+	case HookEventRelease:
+		return "release"
+	}
+	return ""
+}
+
```

Comment:
@lafriks So this turned out to be easier than enums because we expect the API (among possibly other things) to send in slices of strings to map to events, which means enums would need one func to map to a generic event, and another func to map to explicit event. e.g. generic `issues` vs explicit `issue_assign`

This way they are _always_ tied to explicit events and this func just maps them back to a generic event.

I'm not sure I would say this is _cleaner_ but it should avoid breaking change. :sweat_smile:

Reply:
How about a translation table (I mean, in the database):

Instead of:

```text
case HookEventIssues, HookEventIssueAssign, HookEventIssueLabel, HookEventIssueMilestone:
		return "issues"
```

we'd use:

```sql
SELECT name FROM webhook_name WHERE id = 5
```

It would be possible to break down hooks names if so wished, or use the old system. The migration step could set up the old names, and there could be a procedure to change them, but fresh installations could use new names. Would that be overkill?

Then again, there's always `app.ini` 😆

Reply:
I'd rather just break them and be done with it, honestly, though I understand the hesitation not to.  
This is simply to avoid a breaking change while still allowing functionality.

A table would just add to the mess  I think. 😅

Reply:
This is quite fundamental for integrations and would break all existing ci integrations, so I still think it is worth the effort not to break things here

Reply:
Just to clarify, this should prevent the feature from being a breaking change.  
When hooks are delivered, they use this func to map back to a generic event.